steve/util.py

```diff
@@ -41,12 +47,45 @@ class ConfigNotFound(SteveException):
     pass
 
 
+class NoOptionError(SteveException):
+    """Denotes we could not find option in section"""
+    pass
+
+
 class Config(object):
+    """base class for configuration file handling classes
+
+    Each derived class should:
+    - have a file_name class attribute
+    - implement load()
+    - implement get(section, key)
+    - provice string attribute _CONFIG as new file template
+    """
+
+    default_status = 'status.yaml'
+
     def __init__(self):
         self._file_name = None
-
-    def initialise(self):
-        raise NotImplementedError
+        self._project_path = None
+
+    @property
+    def project_path(self):
+        """find the project config file in the current or parent directory"""
+        if self._project_path is None:
+            projectpath = os.getcwd()
+            path = os.path.join(projectpath, self.file_name)
+            if not os.path.exists(path):
+                # if not found try the parent directory for configuration file
+                projectpath = os.path.dirname(projectpath)
+                path = os.path.join(projectpath, self.file_name)
+                if not os.path.exists(path):
+                    raise ConfigNotFound('{0} could not be found.'.format(self.file_name))
+            self._project_path = projectpath
+        return self._project_path
+
+    @property
+    def project_file(self):
```

Comment:
steve.ini resp. steve.yaml depending on the type of project file the project started with (ie. this supports any already created project without having to do a conversion).